app/src/main/java/ch/epfl/skysync/database/Table.kt

```diff
@@ -16,48 +16,65 @@ abstract class Table<M, S : Schema<M>>(
     protected val path: String
 ) {
 
+  /**
+   * Wrap the [run] code in a try/catch block and, if given, call the [onError] callback when an
+   * exception is thrown in [run], in any case throw the exception
+   *
+   * @param onError The callback called when an exception is thrown
+   * @param run The code to run in the try/catch
+   * @throws Exception Any exception thrown in [run]
+   */
+  protected suspend fun <T> withErrorCallback(
+      onError: ((Exception) -> Unit)?,
+      run: suspend () -> T
+  ): T {
+    try {
+      return run()
+    } catch (e: Exception) {
+      if (onError != null) {
+        onError(e)
+      }
+      throw e
+    }
+  }
+
   /**
    * Get an item by ID
    *
    * @param id The id of the item
-   * @param onCompletion Callback called on completion of the operation
    * @param onError Callback called when an error occurs
    */
-  open fun get(id: String, onCompletion: (M?) -> Unit, onError: (Exception) -> Unit) {
-    db.getItem(path, id, clazz, { onCompletion(it?.toModel()) }, onError)
+  open suspend fun get(id: String, onError: ((Exception) -> Unit)? = null): M? {
```

Comment:
What does the open keyword ?

Reply:
It used to declare that a function can be overridden, as methods are by default final in Kotlin.